server/src/main/java/com/sauron/service/CurrentBalanceServiceImpl.java

```diff
@@ -32,29 +32,23 @@ public CurrentBalanceServiceImpl(UserRepository userRepository, RestTemplateBuil
         this.restTemplate = restTemplateBuilder.build();
     }
 
-    //TODO to reconsider after adding some persistence (?) mechanisms to transactions
     @Override
     public BigDecimal getCurrentBalance(final Long userId) {
 
         Collection<BankAccount> userAccounts = userRepository.findById(userId)
                 .map(User::getBankAccounts)
-                .orElse(Collections.emptyList());
+                .orElse(Collections.emptySet());
 
         return userAccounts.stream()
                 .map(acc -> fetchCurrentBalance(acc.getBank(), userId))
                 .reduce(BigDecimal.ZERO, BigDecimal::add);
     }
 
-    private BigDecimal fetchCurrentBalance(Bank bankView, Long userId) {
-        return restTemplate.exchange(
-                createRequestEntity(bankView.getBalanceUrl(), userId),
-                RESPONSE_TYPE)
-                .getBody();
+    private BigDecimal fetchCurrentBalance(Bank bank, Long userId) {
+        Optional<String> apiUrl = bank.getBankApiUrl(GET_BALANCE);
```

Comment:
Personally, I don't like creating local Optional variables if not needed. I would inline it.

Reply:
Change to both services